src/restLevelClient/generateResponseTypes.ts

```diff
@@ -194,7 +197,7 @@ function getResponseHeaderInterfaceDefinition(
         return {
           name: `"${h.header.toLowerCase()}"`,
           ...(description && { docs: [{ description }] }),
-          type: "string",
+          type: primitiveSchemaToType(h.schema),
```

Comment:
After my current PR is merged this will need to take an additional param

`primitiveSchemaToType(h.schema, SchemaContext.Output)`